### DISCUSSION TITLE
Priority cleaning

### DISCUSSION THREAD

**Discussion by dmlemos** [Hypfer/Valetudo#920] [Archive]
The idea to clean zones in a specific order, commanded by the user.

When one opens the zone list, instead of selecting the zones with a checkbox, there would be a button to switch to _priority cleaning_ with numbering. So that the first selected box would become 1, the second 2, and so on.
This way, the robot would clean in the order specified by the user, as opposed to the order that appears in the list.

**Comment by depau** [comment]
This is highly robot dependent, I'd think that most robots will disregard the order that you say and choose the next room to clean based on how far it is from it.

Viomi does this as far as I know, and I would expect the other robots to do the same since it minimizes battery consumption.

All solutions that are along the lines of "start this room, wait for it to finish, start next room" on Valetudo's side are out of scope. You can write an external automation script or rule for that without bloating to the core code - it can be easily done with openHAB without writing a single line of code.

**Reply by dmlemos** [reply]
In the case of the Roborock S5, I notice there is no room optimisation. It just follows the list in order. Which is also quite funny to see him go back and forth, when the rooms are far apart.

I guess maintaining a cleaning queue might be too taxing for the robot's hardware.
I have yet to try the MQTT integration. Thanks for letting me know that it is possible though :)

**Reply by 8times4** [reply]
Chiming in, Viomi also goes by RoomID order, ascending from 1 upwards, as it does not give a damn which room is closer. (In segment cleaning at least)

**Comment by depau** [comment]
In that case it may be worth ensuring the UI sends the segments in the
order they were selected. Though it should not be considered a feature but
rather something that "happens to work on some robot models".

**Reply by dmlemos** [reply]
If the feature is not difficult to implement, it could be of great benefit. In my case, I actually don't want the robot to calculate the optimised path for most of the time.

I'll explain my use case: since I have been working from home, I have to move away from the office, in order for the robot to clean. So that would always be the first room I want it to clean. The same applies if I want it to clean the bathroom before I take a shower, then the living room (so I can go there afterwards), and then clean rest of the house.
For some rooms, it involves me moving some furniture for a deep cleaning before the robot goes there. If I can prioritise it, it's much easier to do this.
There must be plenty of other use cases like mine.

**Reply by Jomik** [reply]
Roborock actually supports ordered cleanup for segments:
https://github.com/Hypfer/Valetudo/commit/25411138688454d43026d07c12e46b5eab5e9269
It should be something that you enable/disable, as sometimes you want it to go in order and sometimes you just want it to be as quick as possible.
I generally want it to be as quick as possible, as I am either working, or out of the house, both meaning that it should just be done ASAP.
But sometimes we want him to start cleaning in the other end of the house, while we eat :)

**Reply by dmlemos** [reply]
Hi @Jomik
The new release just came out #944. I'd like to ask you if this feature is available on the UI?

**Reply by Jomik** [reply]
The new UI is not part of that release, sadly.
Vital parts are still missing, though I am using it as my daily driver.. 😊

**Reply by dmlemos** [reply]
The revamped UI is looking awesome 👍🏽
What I meant to ask is, if the ordered cleanup segments feature is available on the current UI, or through the API only? Perhaps this is a question for the release discussion.

**Reply by Jomik** [reply]
Only through the API currently
